File: python/linear_search.py

def linear_search(list, target):
    """
    Returns the index of the searched if found otherwise -1
    """
    for i in range(0, len(list)):
        if list[i] == target:
            return i
    return -1


def verify(index):
    if index != -1:
        print("Target found at index: ", index)
    else:
        print("Target not found")


numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

result = linear_search(numbers, 4)
verify(result)

result = linear_search(numbers, 7)
verify(result)

File: javascript/linearSearch.js

const linearSearch = (list, target) => {
    for(let i = 0; i < list.length; i++){
        if(list[i] == target){
            return i;
        }
    }
    return -1;
}

const verify = (index) => {
    if(index != -1){
        console.log("Target found at: ", index);
    }
    else{
        console.log("Target not found");
}

}

const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

const target1 = linearSearch(numbers, 12);
verify(target1);

const target2 = linearSearch(numbers, 6);
verify(target2);

